### DISCUSSION TITLE
Repository Event `renamed`/`transferred` Actions

### DISCUSSION THREAD

**Discussion by anton-yurchenko** [community/community#7947] [Actions]
Hey all,
I have paid attention to some weird things about a [repository event](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#repository):

- `renamed` action does not contain a previous repository name, so its impossible to understand which repository was renamed
- `transferred` event is fired only on *arrivals* and not *departures*

Is this going to be changed?
Are there any workarounds for this?

Thanks

**Comment by ethomson** [comment]
Hi @anton-yurchenko - thanks for the feedback.  At the moment, Actions really just uses the webhook data for these particular events.  I'll need to check in with the webhook team and see if they have any workarounds for you, or plans to change this.

**Comment by prudhvigodithi** [comment]
Hey this is same with [issue transfer](https://docs.github.com/en/rest/issues/events?apiVersion=2022-11-28), the event is not recorded on the source repository, but shows on destination repository as `event": "transferred",`, with this even using webhooks on the source repo does not give much info, any workaround for this to get this issue transferred event recorded on source repo?

The [document for transferred issue event ](https://docs.github.com/en/developers/webhooks-and-events/events/issue-event-types#transferred) mentioned as `The issue was transferred to another repository.`, but with this the event is only recorded in destination repo where the issue was already transferred. 

We have an internal Database that needs to be purged for the transferred repo issues and right now there is no way to get the event for issue transfer on the source repo.

One work around is use check for issue API response with code 301, but having to do it across all the existing issues for multiple repos is costing lot of API calls and with limited API's for github the jobs falis
```
{
  "message": "Moved Permanently",
  "url": "https://api.github.com/repos/opensearch-project/opensearch-devops/issues/100",
  "documentation_url": "https://docs.github.com/v3/#http-redirects"
}
```

It would be nice to get the exact transfer event for a single issue on the source repo.
